crates/storage/codecs/src/alloy/eip4844.rs

```diff
@@ -0,0 +1,128 @@
+    pub nonce: u64,
+
+    /// A scalar value equal to the maximum
+    /// amount of gas that should be used in executing
+    /// this transaction. This is paid up-front, before any
+    /// computation is done and may not be increased
+    /// later; formally Tg.
+    pub gas_limit: u64,
+
+    /// A scalar value equal to the maximum
+    /// amount of gas that should be used in executing
+    /// this transaction. This is paid up-front, before any
+    /// computation is done and may not be increased
+    /// later; formally Tg.
+    ///
+    /// As ethereum circulation is around 120mil eth as of 2022 that is around
+    /// 120000000000000000000000000 wei we are safe to use u128 as its max number is:
+    /// 340282366920938463463374607431768211455
+    ///
+    /// This is also known as `GasFeeCap`
+    pub max_fee_per_gas: u128,
+
+    /// Max Priority fee that transaction is paying
+    ///
+    /// As ethereum circulation is around 120mil eth as of 2022 that is around
+    /// 120000000000000000000000000 wei we are safe to use u128 as its max number is:
+    /// 340282366920938463463374607431768211455
+    ///
+    /// This is also known as `GasTipCap`
+    pub max_priority_fee_per_gas: u128,
+
+    /// TODO(debt): this should be removed if we break the DB.
+    /// Makes sure that the Compact bitflag struct has one bit after the above field:
+    /// <https://github.com/paradigmxyz/reth/pull/8291#issuecomment-2117545016>
+    pub placeholder: Option<CompactPlaceholder>,
+
+    /// The 160-bit address of the message call’s recipient.
+    pub to: Address,
+
+    /// A scalar value equal to the number of Wei to
+    /// be transferred to the message call’s recipient or,
+    /// in the case of contract creation, as an endowment
+    /// to the newly created account; formally Tv.
+    pub value: U256,
+
+    /// The accessList specifies a list of addresses and storage keys;
+    /// these addresses and storage keys are added into the `accessed_addresses`
+    /// and `accessed_storage_keys` global sets (introduced in EIP-2929).
+    /// A gas cost is charged, though at a discount relative to the cost of
+    /// accessing outside the list.
+    pub access_list: AccessList,
+
+    /// It contains a vector of fixed size hash(32 bytes)
+    pub blob_versioned_hashes: Vec<B256>,
+
+    /// Max fee per data gas
+    ///
+    /// aka BlobFeeCap or blobGasFeeCap
+    pub max_fee_per_blob_gas: u128,
+
+    /// Unlike other transaction types, where the `input` field has two uses depending on whether
+    /// or not the `to` field is [`Create`](crate::TxKind::Create) or
+    /// [`Call`](crate::TxKind::Call), EIP-4844 transactions cannot be
+    /// [`Create`](crate::TxKind::Create) transactions.
+    ///
+    /// This means the `input` field has a single use, as data: An unlimited size byte array
+    /// specifying the input data of the message call, formally Td.
+    pub input: Bytes,
+}
```

Comment:
Because we need it as it derives `Compact` so this is a trick to be able to implement `Compact` for alloy equivalent structure as long as there is 1:1 compatibility between this definition and alloy's definition

Reply:
same applies to docs here https://github.com/paradigmxyz/reth/pull/10262#discussion_r1739134834b

Reply:
Just fixed that:
- I've removed the doc for all fields
- I've put an explanation on the top for the struct
- I've put a notice (similar to the withdrawals) `Make sure this struct is 1:1 with [`alloy_consensus::transaction::TxEip4844`]` for the reader to refer to the appropriate location in alloy
- As the fields of the structures are not public I didn't repeat the reference to [`alloy_consensus::transaction::TxEip4844`] for each field, which seemed a bit too verbose and repetitive to me. I think that with the notice at the beginning of the structure it's good, wdyt?

Reply:
even better to just remove the `pub` visibility